numpy/ma/core.py

```diff
@@ -3802,7 +3802,10 @@ def __iadd__(self, other):
         else:
             if m is not nomask:
                 self._mask += m
-        ndarray.__iadd__(self._data, np.where(self._mask, 0, getdata(other)))
+        ndarray.__iadd__(
```

Comment:
Another way we might be able to get around the ugliness of those expressions, is to do something like `self._data[~self._mask] += ...`. I have not checked the performance, but that sort of expression does seem to work without raising an `AttributeError`.